### PR TITLE
Us 05 confirmcandidate

### LINKED ISSUE #24: Candidate confirmation
Linked to #5

This story focuses on allowing candidates to be confirmed

### PR DESCRIPTION
**Pull request by Cyphle**
Resolves #24
Beware only taken into account confirmation (no payment nor accomodation are present for the moment)

### CODE REVIEW

File: src/main/java/fr/socrates/Main.java [hunk 1]
```diff
@@ -91,6 +96,23 @@ public static void main(String[] args) {
                     consolePrinter.print(MENU_MESSAGE);
                     choice = scanner.next();
                     break;
+                case SEVEN:
+                    consolePrinter.print(format(candidateService.getRegisteredCandidates()));
+                    consolePrinter.print("Tape l'email du candidat à confirmer");
+                    Optional<Candidate> candidate = candidateService.findCandidateByEmail(scanner.next());
+                    candidate.ifPresent(c -> {
```
Comment (Mounkito) [inline src/main/java/fr/socrates/Main.java]: What is "c"

File: src/main/java/fr/socrates/Main.java [hunk 2]
```diff
@@ -91,6 +96,23 @@ public static void main(String[] args) {
...
+                        boolean confirmation = confirmationService.confirm(c);
+                        System.out.println(confirmation);
```
Comment (Mounkito) [inline src/main/java/fr/socrates/Main.java]: u should use consolePrinter

File: src/main/java/fr/socrates/domain/candidate/CandidateServiceImpl.java [hunk 3]
```diff
@@ -21,4 +23,14 @@ public void add(Candidate candidate) {
...
     }
+
+    @Override
+    public Optional<Candidate> findCandidateByEmail(String email) {
+        return candidateRepository.findByEmail(email);
+    }
+
+    @Override
+    public Optional<Candidate> findCandidate(Candidate candidate) {
+       return candidateRepository.findAll().stream().filter(e -> e.equals(candidate)).findAny();
```
Comment (Mounkito) [inline src/main/java/fr/socrates/domain/candidate/CandidateServiceImpl.java]: What is "e"

File: src/main/java/fr/socrates/domain/attendee/ConfirmationService.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package fr.socrates.domain.attendee;
+
+import fr.socrates.domain.candidate.Candidate;
+
+import java.util.List;
+
+/**
+ * Created by lenovo_13 on 17/08/2017.
```
Comment (udebella) [inline src/main/java/fr/socrates/domain/attendee/ConfirmationService.java]: On peut se passer de ce genre de commentaires :)

File: src/test/java/fr/socrates/domain/attendee/ConfirmationServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ConfirmationServiceImplTest {
+  private CandidateService candidateService;
+  private ConfirmationServiceImpl confirmationServiceImpl;
+
+  @Before
+  public void setUp() throws Exception {
+    candidateService = Mockito.spy(new CandidateServiceImpl(null));
```
Comment (Mounkito) [inline src/test/java/fr/socrates/domain/attendee/ConfirmationServiceImplTest.java]: u shoud use a simple Mock

**Review by Mounkito — CHANGES_REQUESTED** [review]
I think some changes are necessary

**Review by udebella — CHANGES_REQUESTED** [review]

**Review by bassail — APPROVED** [review]
Nice Job

**Review by Mounkito — APPROVED** [review]

**Review by udebella — APPROVED** [review]

**Pull request merged**